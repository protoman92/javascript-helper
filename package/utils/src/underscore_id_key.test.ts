import underscoreIDKeys from "./underscore_id_key";

describe("Underscore ID key", () => {
  it("Replace ID key", async () => {
    expect(
      underscoreIDKeys(
        {
          id: 1,
          a: { id: 2, b: { id: 3, c: "Big deal" }, c: "No big deal" },
          c: `"id":`,
          d: "id:",
          ignored: "e",
        },
        (key, value) => (key === "ignored" ? undefined : value)
      )
    ).toMatchSnapshot();
  });
});
