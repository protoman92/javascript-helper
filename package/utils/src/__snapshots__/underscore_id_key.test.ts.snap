// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Underscore ID key Replace ID key 1`] = `
{
  "_id": 1,
  "a": {
    "_id": 2,
    "b": {
      "_id": 3,
      "c": "Big deal",
    },
    "c": "No big deal",
  },
  "c": ""id":",
  "d": "id:",
}
`;
